### PR TITLE
feat: Add default TabBar component

### PR DESCRIPTION
**Pull request by paulomeurerzup**
### Related Issues

#1306

### Description and Example

Adds default TabBar and PageView components. It isn't possible to style the components yet.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: flutter/beagle/lib/beagle_initializer.dart [hunk 1]
```diff
@@ -59,4 +63,8 @@ class BeagleInitializer {
   static BeagleService getService() {
     return _service;
   }
+
+  static DesignSystem get designSystem {
+    return _designSystem;
+  }
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/beagle_initializer.dart:69]:
```suggestion
  static DesignSystem get designSystem => _designSystem;
```

File: flutter/beagle_components/lib/beagle_image.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+abstract class ImagePath {
+  ImagePath._();
+
+  factory ImagePath.local(String mobileId) = LocalImagePath;
+
+  factory ImagePath.remote(String url) = RemoteImagePath;
+
+  factory ImagePath.fromJson(Map<String, dynamic> json) {
+    if (json['_beagleImagePath_'] == 'local') {
```
Comment (gabrielleonardozup) [inline flutter/beagle_components/lib/beagle_image.dart:25]: It is a good idea insert this key string values into constants.

File: flutter/beagle_components/lib/beagle_image.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+class LocalImagePath extends ImagePath {
+  LocalImagePath(this.mobileId) : super._();
+
+  final String mobileId;
+}
+
+class RemoteImagePath extends ImagePath {
+  RemoteImagePath(this.url) : super._();
+  final String url;
+// TODO: placeholder
```
Comment (matheusribeirozup) [inline flutter/beagle_components/lib/beagle_image.dart:42]: Why was the placeholder not implemented?

Comment (paulomeurerzup) [reply flutter/beagle_components/lib/beagle_image.dart:42]: It's gonna be implemented with the Image component

File: flutter/beagle_components/lib/beagle_page_view.dart [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    super.initState();
+    _pageController = PageController(initialPage: widget.currentPage);
+    _pageController.addListener(() {
+      // Here we're trying to do a smooth transition between pages
+      final isScrollingLeft = _pageController.page < _lastPage;
+      final pageFraction =
+          _pageController.page - _pageController.page.truncate();
+      final page = _pageController.page.round();
+      if (isScrollingLeft) {
+        if (pageFraction < 0.10) {
```
Comment (gabrielleonardozup) [inline flutter/beagle_components/lib/beagle_page_view.dart:53]: A suggestion here is insert this fraction numbers into constants with representative names, it helps to improve code readability

Comment (gabrielleonardozup) [reply flutter/beagle_components/lib/beagle_page_view.dart:53]: Another suggestion here is separate if and else case in two separates methods to create a context for them. Also helps code readability.

File: flutter/sample/lib/main.dart [hunk 5]
```diff
@@ -77,20 +84,44 @@ class _BeagleSampleApp extends State<BeagleSampleApp> {
...
-            ? const BeagleRemoteView(route: '/lazy.json')
+            ? const BeagleRemoteView(route: '/beagle_lazy')
             : const Center(
                 child: Text('Not ready yet!'),
               ),
       ),
     );
   }
+
+  void handleAppBarMenuOption(String menuOption) {
```
Comment (matheusribeirozup) [inline flutter/sample/lib/main.dart:119]:
```suggestion
  void _handleAppBarMenuOption(String menuOption) {
```

File: flutter/beagle/lib/beagle_initializer.dart (hunk 1 above)
Comment (paulomeurerzup) [reply flutter/beagle/lib/beagle_initializer.dart]: done

File: flutter/beagle_components/lib/beagle_page_view.dart (hunk 4 above)
Comment (paulomeurerzup) [reply flutter/beagle_components/lib/beagle_page_view.dart]: done

File: flutter/sample/lib/main.dart (hunk 5 above)
Comment (paulomeurerzup) [reply flutter/sample/lib/main.dart]: done

File: flutter/beagle_components/lib/beagle_image.dart (hunk 2 above)
Comment (paulomeurerzup) [reply flutter/beagle_components/lib/beagle_image.dart]: done

**Review by matheusribeirozup — APPROVED** [review]

File: flutter/beagle_components/lib/default_components.dart [hunk 6]
```diff
@@ -32,6 +35,8 @@ final Map<String, ComponentBuilder> defaultComponents = {
   'beagle:textInput': beagleTextInputBuilder(),
   'beagle:button': beagleButtonBuilder(),
   'beagle:lazycomponent': beagleLazyComponentBuilder(),
+  'beagle:tabbar': beagleTabBarBuilder(),
```
Comment (gabrielleonardozup) [inline flutter/beagle_components/lib/default_components.dart:38]: pattern here is different from line 35, is that correct?

**Review by gabrielleonardozup — APPROVED** [review]

**Pull request merged**